packages/digitalriver-integration/src/digitalriver-payment-strategy.ts

```diff
@@ -43,32 +44,29 @@ export default class DigitalRiverPaymentStrategy implements PaymentStrategy {
     private _digitalRiverCheckoutData?: DigitalRiverInitializeToken;
     private _unsubscribe?: () => void;
     private _digitalRiverInitializeOptions?: DigitalRiverPaymentInitializeOptions;
-
     constructor(
-        private _store: CheckoutStore,
-        private _paymentMethodActionCreator: PaymentMethodActionCreator,
-        private _orderActionCreator: OrderActionCreator,
-        private _paymentActionCreator: PaymentActionCreator,
-        private _storeCreditActionCreator: StoreCreditActionCreator,
+        private _paymentIntegrationService: PaymentIntegrationService,
         private _digitalRiverScriptLoader: DigitalRiverScriptLoader,
-        private _billingAddressActionCreator: BillingAddressActionCreator,
     ) {}
 
-    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+    async initialize(
+        options: PaymentInitializeOptions & WithDigitalRiverPaymentInitializeOptions,
+    ): Promise<void> {
         this._digitalRiverInitializeOptions = options.digitalriver;
 
-        const paymentMethod = this._store
+        const paymentMethod = this._paymentIntegrationService
             .getState()
-            .paymentMethods.getPaymentMethodOrThrow(options.methodId);
-        const { publicKey, paymentLanguage: locale } = paymentMethod.initializationData;
+            .getPaymentMethodOrThrow<DigitalRiverInitializationData>(options.methodId);
+        const { publicKey, paymentLanguage: locale } =
+            paymentMethod.initializationData as DigitalRiverInitializationData;
```

Comment:
We need to get rid of using the as operator everywhere. Instead of it we have to create type guard in case where it is needed. 

But in your case there is no need, because you are using generic and `initializationData` alredy has an appropriate interface.

Just use it something like that: `const { ... } = paymentMethod.initializationData || {};`

Reply:
If I do it this way: `const { ... } = paymentMethod.initializationData || {};` when I try to use publicKey later in the code, I have error ```
Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.
```